### PR TITLE
Fix zid registry false positives from Zod 4 parent inheritance

### PR DESCRIPTION
**Pull request by mikecann**
## Summary

- Move the `_zidRegistry` lookup in `zodToConvexCommon` into the `$ZodCustom` branch (the type `zid()` produces), rather than checking it at the top of the function before any type-specific `instanceof` checks. This ensures type-specific handlers like `$ZodString`, `$ZodNumber`, etc. always take priority.
- This is a defensive improvement: since `zid()` always produces a `$ZodCustom`, there's no reason to check the zid registry for other types. Scoping the check to `$ZodCustom` makes the code more precise and prevents any future issues if Zod's internal `_zod.parent` chain were to cause registry lookups to match unexpected types.
- Add regression tests for `zid().describe()` and `z.string().describe()` to ensure derived schemas are handled correctly.

## Test plan
- [x] All 150 zodToConvex tests pass
- [x] All zod4-related tests pass
- [x] Full package test suite passes

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive tests for schema-ID handling, covering derived schemas, parent-chain inheritance, primitives, object fields, and edge cases to prevent regressions.

* **Refactor**
  * Adjusted schema conversion logic to limit registry lookups to the custom-schema path, fixing misidentification across derived schemas.

* **Style**
  * Reflowed several type/generic declarations across packages for consistent formatting (no behavioral changes).

### CODE REVIEW

File: packages/convex-helpers/react/sessions.ts [hunk 1]
```diff
@@ -269,8 +269,9 @@ type SessionMutation<Mutation extends FunctionReference<"mutation">> = (
 ) => Promise<FunctionReturnType<Mutation>>;
 
 // Similar to ReactMutation, but with a sessionId parameter.
-interface ReactSessionMutation<Mutation extends FunctionReference<"mutation">>
```
Comment (Nicolapps) [inline packages/convex-helpers/react/sessions.ts]: These look like reformatting changes that are unrelated to the contents of the PR. Could you please revert them?

Comment (mikecann) [reply packages/convex-helpers/react/sessions.ts]: ye the prettier checks are failing without these changes, opus tells me that these formatting issues are on main too

Comment (Nicolapps) [reply packages/convex-helpers/react/sessions.ts]: You’re right, they seem to be caused by the Prettier update in 50ceecf7071d99daf82f0eda5d2de2d074e83ca0. I’m fixing them on `main`
